aspnetcore/migration/22-to-30.md

```diff
@@ -75,6 +75,108 @@ This article explains how to update an existing ASP.NET Core 2.2 project to ASP.
+        .ConfigureWebHostDefaults(webBuilder =>
+        {
+            webBuilder.UseKestrel(serverOptions =>
+            {
+                // Set properties and call methods on options
+            })
+            .UseIISIntegration()
+            .UseStartup<Startup>();
+        })
+        .Build();
+
+    host.Run();
+}
+```
+
+### Connection Middleware replaces Connection Adapters
+
+Connection Adapters (<xref:Microsoft.AspNetCore.Server.Kestrel.Core.Adapter.Internal.IConnectionAdapter>) has been removed from Kestrel. Replace Connection Adapters with Connection Middleware (similar to HTTP Middleware in the ASP.NET Core pipeline but for lower-level connections). HTTPS and connection logging have been moved from Connection Adapters to Connection Middleware, so those extension methods should continue to work as in previous versions of ASP.NET Core. For more information, see [the TlsFilterConnectionHandler example in the ListenOptions.Protocols section of the Kestrel article](/aspnet/core/fundamentals/servers/kestrel?view=aspnetcore-3.0#listenoptionsprotocols).
+
+### Transport abstractions moved and made public
+
+The Kestrel transport layer has been exposed as a public interface in `Connections.Abstractions`. As part of these updates:
+
+* `Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions` and associated types have been removed.
+* <xref:Microsoft.AspNetCore.Server.Kestrel.KestrelServerOptions.NoDelay> was moved from <xref:Microsoft.AspNetCore.Server.Kestrel.Core.ListenOptions> to the transport options.
+* <xref:Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.Internal.SchedulingMode> was removed from <xref:Microsoft.AspNetCore.Server.Kestrel.KestrelServerOptions>.
+
+For more information, see the following GitHub resources:
+
+* [Client/server networking abstractions (aspnet/AspNetCore #10308)](https://github.com/aspnet/AspNetCore/issues/10308)
+* [Implement new bedrock listener abstraction and re-plat Kestrel on top (aspnet/AspNetCore #10321)](https://github.com/aspnet/AspNetCore/pull/10321)
+
+### Kestrel Request trailer headers
+
+For apps that target earlier versions of ASP.NET Core, Kestrel adds HTTP/1.1 chunked trailer headers into the request headers collection when the request body is read to the end. This causes some concerns about ambiguity between headers and trailers, so the trailers have been moved to a new collection (`RequestTrailerExtensions`) in 3.0.
+
+HTTP/2 request trailers, which are unavailable in ASP.NET Core 2.2, are now available in 3.0 as `RequestTrailerExtensions`. New request extension methods are present to access these trailers. HTTP/2 trailers are available as soon as they're received from the client, but the client doesn't send the trailers until the entire request body has been at least buffered by the server. You may need to read the request body to free up buffer space. Trailers are always available if the request body is read to the end, as the trailers mark the end of the body.
```

Comment:
I wouldn't even mention that the HTTP/2 trailers might be available before reading the entire request body, because even if that's true, it's unreliable. I'd go with the following.

```suggestion
HTTP/2 request trailers, which are unavailable in ASP.NET Core 2.2, are now available in 3.0 as `RequestTrailerExtensions`. New request extension methods are present to access these trailers. As with HTTP/1.1, trailers are available after the request body is read to the end.
```